### PR TITLE
Implement TypeAdapter using Schema instead of reflection.

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: wire-schema/src/main/java/com/squareup/wire/schema/SchemaTypeAdapterFactory.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    @Override public int dataSize(Object value) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override public void writeData(Object value, ProtoWriter writer) throws IOException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override public Object read(ProtoReader reader) throws IOException {
+      Integer value = TypeAdapter.INT32.read(reader);
```
Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/SchemaTypeAdapterFactory.java]: `UINT32`

**Comment by JakeWharton** [comment]
LGTM. Nice to see the formalized concepts in the runtime (message and type adapters) are at least functional in other environments of serialization. Bodes well for future code-generation of message adapters.

File: wire-schema/src/main/java/com/squareup/wire/schema/SchemaTypeAdapterFactory.java (hunk 1 above)
Comment (JakeWharton) [inline wire-schema/src/main/java/com/squareup/wire/schema/SchemaTypeAdapterFactory.java]: (For that person who specifies an enum tag of 2147483648)

Comment (swankjesse) [inline wire-schema/src/main/java/com/squareup/wire/schema/SchemaTypeAdapterFactory.java]: Good find. Fixed.

**Comment by danrice-square** [comment]
Great stuff!

**Pull request merged**